### DISCUSSION TITLE
AvroProducer and AvroConsumer

### DISCUSSION THREAD

**Discussion by Byzas19** [confluentinc/confluent-kafka-python#1288] [Q&A]
Good morning,

I've noticed that in some documentation AvroProducer and AvroConsumer are marked as "[Legacy](https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#avroproducer-legacy)"  But in other places (like [here ](https://pypi.org/project/confluent-kafka/)and the README of this repo), they are not.   

I have a scenario where the AvroProducer couple be potentially helpful but I don't want to use it if it is considered a legacy component that will be someday phased out. Any thoughts or suggestions?

**Comment by jpweng** [comment]
What does it mean that a class (or anything else) is marked "Legacy"? Is it not maintained anymore and its usage should be avoided?
